tools/waveica/waveica_wrapper.R

```diff
@@ -10,25 +10,28 @@ waveica <- function(
 ) {
 
     # get input from the Galaxy, preprocess data
-    data <- read.csv(data, header = TRUE, row.names = "sampleName")
+    data <- read.csv(data, header = TRUE)
+
+    required_columns <- c("sampleName", "class", "sampleType", "injectionOrder", "batch")
+    if (anyNA(data)) {
+        stop("Error: dataframe cannot contain NULL values!
+Make sure that your dataframe does not contain empty cells")
+    }
+    else if (!all(required_columns %in% colnames(data))) {
+        stop("Error: missing metadata!
+Make sure that the following columns are present in your dataframe: [sampleName, class, sampleType, injectionOrder, batch]")
+    }
 
     # sort data by injection order
     data <- data[order(data$injectionOrder, decreasing = FALSE), ]
```

Comment:
Does this injection order also reflect the batch number? Because I assume that the ordering should be batch > injectionOrder, otherwise all first samples of all batches will be on the top and all last samples on the bottom etc.

Or is this the desired behaviour?

Reply:
This is an excellent point! I assumed that the injection order would be batch-independent, i.e., the numbering would go `from 1 to the number of samples` as it was in the test data of the `WaveICA` package. But it wouldn't work if the numbering repeated for every batch. I accounted for that in 49f4b2a.